lang/java/avro/src/main/java/org/apache/avro/file/CodecFactory.java

```diff
@@ -40,7 +42,9 @@
 public abstract class CodecFactory {
   /** Null codec, for no compression. */
   public static CodecFactory nullCodec() {
-    return NullCodec.OPTION;
+    // we can not reference NullCodec.OPTION because the static
```

Comment:
The unit test uncovered the fact that accessing this field here results in 'null' since we have a circular dependency in static initialization.